core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/StreamBindingService.java

```diff
@@ -92,7 +93,32 @@
 
   @PreAuthorize("hasPermission(#streamBinding, 'DELETE')")
   public void delete(StreamBinding streamBinding) {
-    throw new UnsupportedOperationException();
+    val existing = unsecuredGet(streamBinding.getKey());
+    if (!existing.isPresent()) {
+      throw new ValidationException("Can't delete " + streamBinding.getKey() + " because it doesn't exist");
+    }
+    handlerService.handleDelete(streamBinding);
+    findAllAndDelete(streamBinding.getKey());
+    streamBindingRepository.delete(streamBinding);
+  }
+
+  private void findAllAndDelete(StreamBindingKey key) {
```

Comment:
Did you mean to make this a private method? In the other services, this is something like
```suggestion
  @PreAuthorize("returnObject.isPresent() ? hasPermission(returnObject, 'DELETE') : true")
  public void findAllAndDelete(StreamBindingKey key) {
```